cleanlab/outlier.py

```diff
@@ -514,6 +526,15 @@ def _get_ood_predictions_scores(
         ood_predictions_scores = 1.0 - get_normalized_entropy(pred_probs)
     elif method == "least_confidence":
         ood_predictions_scores = pred_probs.max(axis=1)
+    elif method == "gen":
+        if pred_probs.shape[1] < M:
+            warnings.warn(
+                "GEN with the default hyperparameter settings is intended for datasets with at least 100 classes. You can change the params['M'] according to the number of classes in your dataset.",
+                UserWarning,
+            )
+        probs = softmax(pred_probs, axis=1)
+        probs_sorted = np.sort(probs, axis=1)[:,-M:]
+        ood_predictions_scores = -np.sum(probs_sorted**gamma * (1 - probs_sorted)**(gamma), axis=1)
```

Comment:
please confirm:

You are transforming the GEN scores (posthoc) to ensure they lie in 0-1 and smaller values == datapoints that are more severe outliers

Reply:
seems fine to me. Please add:

1. a comment in the code clarifying that you're doing this
2. a unit test that ensures all values are in 0-1 range, and smallest values are the outliers in a toy dataset

Reply:
Yes I have already added the unit test to ensure all values are in 0-1 range. And for the smallest values things, I saw your another comment. Yes the ood score will be 1 for both `entropy` and `least_confidence` methods because the prediction confidence will be 1 for that datapoint. I tried using the mean of three means of data distribution and it works well for me.